File: src/main/java/org/apache/fineract/client/services/SelfDividendApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface SelfDividendApi {
  /**
   * 
   * 
   * @param productId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("shareproduct/{productId}/dividend")
  Observable<String> createDividendDetail(
    @retrofit2.http.Path("productId") Long productId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param productId  (required)
   * @param dividendId  (required)
   * @return Observable&lt;String&gt;
   */
  @DELETE("shareproduct/{productId}/dividend/{dividendId}")
  Observable<String> deleteDividendDetail(
    @retrofit2.http.Path("productId") Long productId, @retrofit2.http.Path("dividendId") Long dividendId
  );

  /**
   * 
   * 
   * @param productId  (required)
   * @param offset  (optional)
   * @param limit  (optional)
   * @param orderBy  (optional)
   * @param sortOrder  (optional)
   * @param status  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("shareproduct/{productId}/dividend")
  Observable<String> retrieveAll39(
    @retrofit2.http.Path("productId") Long productId, @retrofit2.http.Query("offset") Integer offset, @retrofit2.http.Query("limit") Integer limit, @retrofit2.http.Query("orderBy") String orderBy, @retrofit2.http.Query("sortOrder") String sortOrder, @retrofit2.http.Query("status") Integer status
  );

  /**
   * 
   * 
   * @param dividendId  (required)
   * @param productId  (required)
   * @param offset  (optional)
   * @param limit  (optional)
   * @param orderBy  (optional)
   * @param sortOrder  (optional)
   * @param accountNo  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("shareproduct/{productId}/dividend/{dividendId}")
  Observable<String> retrieveDividendDetails(
    @retrofit2.http.Path("dividendId") Long dividendId, @retrofit2.http.Path("productId") Long productId, @retrofit2.http.Query("offset") Integer offset, @retrofit2.http.Query("limit") Integer limit, @retrofit2.http.Query("orderBy") String orderBy, @retrofit2.http.Query("sortOrder") String sortOrder, @retrofit2.http.Query("accountNo") String accountNo
  );

  /**
   * 
   * 
   * @param productId  (required)
   * @param dividendId  (required)
   * @param command  (optional)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("shareproduct/{productId}/dividend/{dividendId}")
  Observable<String> updateDividendDetail(
    @retrofit2.http.Path("productId") Long productId, @retrofit2.http.Path("dividendId") Long dividendId, @retrofit2.http.Query("command") String command, @retrofit2.http.Body String body
  );

}

File: src/main/kotlin/org/openapitools/client/apis/SelfDividendApi.kt
package org.openapitools.client.apis

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query


interface SelfDividendApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param productId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/shareproduct/{productId}/dividend")
    suspend fun createDividendDetail(
        @Path("productId") productId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param productId
     * @param dividendId
     * @return [kotlin.String]
     */
    @DELETE("v1/shareproduct/{productId}/dividend/{dividendId}")
    suspend fun deleteDividendDetail(
        @Path("productId") productId: Long,
        @Path("dividendId") dividendId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param productId
     * @param offset  (optional)
     * @param limit  (optional)
     * @param orderBy  (optional)
     * @param sortOrder  (optional)
     * @param status  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/shareproduct/{productId}/dividend")
    suspend fun retrieveAll39(
        @Path("productId") productId: Long,
        @Query("offset") offset: Int? = null,
        @Query("limit") limit: Int? = null,
        @Query("orderBy") orderBy: String? = null,
        @Query("sortOrder") sortOrder: String? = null,
        @Query("status") status: Int? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param dividendId
     * @param productId
     * @param offset  (optional)
     * @param limit  (optional)
     * @param orderBy  (optional)
     * @param sortOrder  (optional)
     * @param accountNo  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/shareproduct/{productId}/dividend/{dividendId}")
    suspend fun retrieveDividendDetails(
        @Path("dividendId") dividendId: Long,
        @Path("productId") productId: Long,
        @Query("offset") offset: Int? = null,
        @Query("limit") limit: Int? = null,
        @Query("orderBy") orderBy: String? = null,
        @Query("sortOrder") sortOrder: String? = null,
        @Query("accountNo") accountNo: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param productId
     * @param dividendId
     * @param command  (optional)
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/shareproduct/{productId}/dividend/{dividendId}")
    suspend fun updateDividendDetail(
        @Path("productId") productId: Long,
        @Path("dividendId") dividendId: Long,
        @Query("command") command: String? = null,
        @Body body: String? = null
    ): String

}
